src/gf.c

```diff
@@ -2327,24 +2367,30 @@ jl_value_t *jl_fptr_const_return(jl_value_t *f, jl_value_t **args, uint32_t narg
 
 jl_value_t *jl_fptr_args(jl_value_t *f, jl_value_t **args, uint32_t nargs, jl_code_instance_t *m)
 {
-    jl_fptr_args_t invoke = jl_atomic_load_relaxed(&m->specptr.fptr1);
+    jl_fptr_args_t invoke = jl_atomic_load_acquire(&m->specptr.fptr1);
     while (1) {
```

Comment:
Is this loop necessary anymore? By the time we invoke `jl_fptr_args`, we should have already set the fptr and therefore the lookup must always succeed right?

Reply:
Yes, the loop lets us use cheaper relaxed loads most of the time. But I think this particular issue here is already fixed on master now?

Reply:
I mean in the sense that if we're calling `jl_fptr_args` at all, then we know that the load from specptr should never be null, and therefore we can delete the loop. Is this a valid claim?

Reply:
I think the `invoke` would likely have to be an acquire load for that to be valid, since we aren't directly dereferencing it to get the specptr to be able to assume it provides consume ordering. Are you suggesting that we should just use acquire loads for the former?

Regardless, I don't see any point in making this an acquire load, since it doesn't "acquire" any writes. It only needs it in the loop, to enforce forward progress there on the compiler, not because we need acquire semantics in the processor.

Reply:
Yeah I think acquire loads on the invoke pointer should be better, since the semantics I'm looking for here are that once invoke is acquired, all subsequent reads are valid wrt the loaded invoke. That lets the initial load here be relaxed and drops the loop as well.

Reply:
Let's go with that then. It is not like this dispatch case is that fast anyways to worry about an extra acquire load right now.